blob/example_openbucket_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blob_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"gocloud.dev/blob"
+	_ "gocloud.dev/blob/fileblob"
+)
+
+func ExampleOpenBucket() {
+	// Connect to a bucket using a URL.
+	// This example uses the file-based implementation, which registers for
+	// the "file" scheme.
+	dir, cleanup := newTempDir()
+	defer cleanup()
+
+	ctx := context.Background()
+
+	// The blob package is designed for using with a specific provider package.
+	// Here blob.OpenBucket needs a provider that implements blob.BucketURLOpener
+	// to work. In this example, we use blob/fileblob, so we need to add a blank
+	// import line to register the fileblob provider:
+	// import _ "gocloud.dev/blob/fileblob"
+	if _, err := blob.OpenBucket(ctx, "file:///nonexistentpath"); err == nil {
+		log.Fatal("Expected an error opening nonexistent path")
+	}
+	fmt.Println("Got expected error opening a nonexistent path")
+
+	// Ensure the path has a leading slash; fileblob ignores the URL's
+	// Host field, so URLs should always start with "file:///". On
```

Comment:
I'm not sure "should always start with ///" is accurate. On Linux, for absolute paths it is true, but you could theoretically pass a relative path.

How about:

var url string
if os.PathSeparator == "/" {
  // Fileblob ignores the "Host" field of the URL; the directory
  // is the URL path. So, URLs for absolute paths like the one we're
  // using here will start with "file:///".
  url = path.Join("file://", dir)
} else {
  // On Windows, we need to convert "\\" to "/" for use in a URL, and
  // add a leading "/". Fileblob will strip the leading "/" and
  // convert the "/" back to "\\".
  url = "file:///" + filepath.ToSlash(dir)
}
```

WDYT?

Reply:
The current way we handle this will ignore the leading `.`, `~` as we only use `URL.Path`. If we think this is something we want to support, we might want to adjust the function behavior.

Reply:
Thinking this over, it can be quite confusing and it's not really that important here for this `blob`-level example.

How about we move this discussion to the `fileblob/example_test` and `fileblob/fileblob_test`? I am not sure what you mean by the leading "." -- it might be good to add a testcase for that to be more explicit about it.

Meanwhile, we can change this example to either use `memblob` (URL will be `mem://`) or `gcsblob` (removing `// Output` line so example only compiles).

Reply:
Changed to use memblob since this example is to demonstrate open by URL.

Filed #1425 for more investigation.